lib/resources/users.js

```diff
@@ -55,7 +55,7 @@ module.exports = (service, endpoint) => {
         .orElseGet(() => Users.emailEverExisted(body.email)
           .then((existed) => ((existed === true)
             ? mail(body.email, 'accountResetDeleted')
-            : resolve())))
+            : (isTrue(query.invalidate) ? reject(Problem.user.insufficientRights()) : resolve()))))
```

Comment:
I'm wondering whether the `isTrue(query.invalidate)` check should come before the `Users.emailEverExisted()` query. If the check came right after `orElseGet()`, that'd be similar to how it's positioned after `map()` above. Right now, since `isTrue(query.invalidate)` isn't checked if `existed` is `true`, it'd be still possible for an unauthed user to use this endpoint to determine whether an email was deleted (if the email was deleted, it'd be a 200 instead of a 403). That's probably not a big deal, but it'd be nice to have this endpoint not provide any information at all to a user who doesn't have sufficient auth (for example, Chelsea in testing).

I'm also wondering what the right behavior is if the authed user is a sidewide administrator (like Alice in testing). I'm thinking that Alice shouldn't receive a 403 here, since the issue isn't really that Alice has insufficient rights. I think it'd be fine for Alice to continue to receive a 200; we just want a user without sufficient auth to receive a 403 (so that they can't gain information). Instead of returning `reject(Problem.user.insufficientRights())`, maybe we could do an auth check in `orElseGet()` that's similar to what's done above in `map()`.

The auth check in `map()` is `auth.canOrReject('user.password.invalidate', user.actor)`, but of course there's no `user` in `orElseGet()`. But I'm thinking it might work to specify `User.species` as the actee when you complete the auth check in `orElseGet()`.

Reply:
Incorporated your feedback. Can you please rerun the build, I think it failed because two builds where running simultaneously